### PR TITLE
Containerized CI and expansion to trigger external tests

### PR DESCRIPTION
**Pull request by mlee03**
**Description**
In this PR, the CI workflow is changed to run in a [container](https://github.com/orgs/NOAA-GFDL/packages/container/package/miniforge).  The image is built from a base _miniforge image_.  GNU compilers, netCDF and HDF5 libraries, and MPICH have been installed.  The default python version is set to 3.11.7.

Files `fv3_translate_tests.yaml`, `pace_tests.yaml`, and `shield_tests.yaml`  have been added to run the CI workflow in to test NDSL changes in pyFV3, pace, and pySHiELD.  

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included

### CODE REVIEW

**Comment by mlee03** [comment]
Depends on pyFV3 [24](https://github.com/NOAA-GFDL/PyFV3/pull/24)

File: .github/workflows/fv3_unit_tests.yaml
Comment (FlorianDeconinck) [inline .github/workflows/fv3_unit_tests.yaml:1]: Rename to `FV3 Translate tests` which are not unit tests, so it's clear that this will run numerical regression.

**Review by fmalatino — DISMISSED** [review]
Approved pending requested changes. To be merged after [PR 53](https://github.com/NOAA-GFDL/NDSL/pull/53)

File: .github/workflows/fv3_unit_tests.yaml
Comment (mlee03) [reply .github/workflows/fv3_unit_tests.yaml]: Done!

**Comment by mlee03** [comment]
I apologize to everyone, I decided to push reusable workflow changes to this existing PR, the PR has changed since last perusal.

**Review by fmalatino — CHANGES_REQUESTED** [review]
After conflicts are resolved, good to merge

**Review by fmalatino — COMMENTED** [review]
To be reviewed after [PR 25](https://github.com/NOAA-GFDL/PyFV3/pull/25#event-13368723004) in `pyFV3`

**Review by oelbert — APPROVED** [review]
lgtm

**Review by FlorianDeconinck — APPROVED** [review]
LGTM

**Pull request merged**
